src/Datadog.Trace/Logging/LibLogCorrelationIdentifierScopeSubscriber.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+
+namespace Datadog.Trace.Logging
+{
+    internal class LibLogCorrelationIdentifierScopeSubscriber : IDisposable
+    {
+        // Keep track of the active Scope (and its corresponding logging contexts)
+        // so if another Scope is deactivated (not the active Scope), no
+        // changes are made
+        private Scope _activeScope;
+        private IDisposable _activeTraceContext;
+        private IDisposable _activeSpanContext;
+
+        public LibLogCorrelationIdentifierScopeSubscriber(IScopeManager scopeManager)
+        {
+            scopeManager.ScopeActivated += OnScopeActivated;
+            scopeManager.ScopeDeactivated += OnScopeDeactivated;
+        }
+
+        public void OnScopeActivated(object sender, ScopeEventArgs scopeEventArgs)
+        {
+            // Dispose of the previous contents since that scope is no longer active
+            _activeTraceContext?.Dispose();
+            _activeSpanContext?.Dispose();
+
+            _activeScope = scopeEventArgs.Scope;
+            _activeTraceContext = LogProvider.OpenMappedContext(CorrelationIdentifier.TraceIdKey, CorrelationIdentifier.TraceId, destructure: false);
```

Comment:
I was eager to use the CorrelationIdentifier API's since that was newly implemented and it would go to the static Tracer instance to get the active TraceId and SpanId, but maybe we can just extract the information from the event args passed into the method. Do you have a preference on which approach to use?

Reply:
I generally prefer passing as arguments. It's my opinion that while we rely on statics, we should keep the coupling to them to a minimum.

Reply:
I think we should use the event args passed into the method in case this event is raised by a different tracer that's not `Tracer.Default` (which is what `CorrelationIdentifier` relies on).

Reply:
That possibility hadn't even occurred to me. Thanks for the feedback guys!